hooks/useRailgunTx.tsx

```diff
@@ -0,0 +1,188 @@
+import { useToken } from "@/contexts/TokenContext";
+import useShieldPrivateKey from "@/hooks/useShieldPrivateKey";
+import {
+  gasEstimateForShield,
+  gasEstimateForShieldBaseToken,
+  populateShield,
+  populateShieldBaseToken,
+} from "@railgun-community/quickstart";
+import {
+  RailgunERC20AmountRecipient,
+  RailgunERC20Amount,
+  NetworkName,
+  TransactionGasDetailsSerialized,
+  EVMGasType,
+  NETWORK_CONFIG,
+  deserializeTransaction,
+} from "@railgun-community/shared-models";
+import { ethers } from "ethers";
+import { getAddress, parseUnits } from "ethers/lib/utils.js";
+import { useAccount, useSigner } from "wagmi";
+import { useProvider } from "wagmi";
+
+const useRailgunTx = () => {
+  const { data: signer } = useSigner();
+  const { address } = useAccount();
+  const { getShieldPrivateKey } = useShieldPrivateKey();
+  const network = NetworkName.EthereumGoerli;
+  const { weth } = useToken();
+  const provider = useProvider();
+
+  const shield = async (args: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    if (args.tokenAddress === getAddress(`0x${"e".repeat(40)}`))
+      return shieldBaseToken(args);
+    return shieldToken(args);
+  };
+
+  const shieldBaseToken = async ({
+    tokenAmount,
+    tokenDecimals,
+    recipient,
+  }: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    // Public wallet to shield from.
+    const fromWalletAddress = address as `0x{string}`;
+
+    // The shieldPrivateKey enables the sender to decrypt
+    // the receiver's address in the future.
+    const shieldPrivateKey = await getShieldPrivateKey();
+
+    const wrappedERC20Amount: RailgunERC20Amount = {
+      tokenAddress: weth!, // wETH
+      amountString: parseUnits(tokenAmount!, tokenDecimals).toHexString(), // hexadecimal amount
+    };
+
+    const { gasEstimateString, error: err } =
```

Comment:
Similarly, does that mean `gasEstimateString` is also a hex string, and not a decimal string? Accepting inputs as ethers.js' `BigNumberish` type and returning `BigNumbers` or `bigint` would be a much nicer SDK interface IMO

Reply:
good call! currently inputs must be hex strings.